### PR TITLE
Issue #341: add Crate support

### PR DESCRIPTION
**Pull request by inge4pres**
This PR introduces an additional dialect to support crate.io as per [issues/341](https://github.com/go-gorp/gorp/issues/341)

### CODE REVIEW

**Comment by nelsam** [comment]
Apologies for the 6 month delay; this looks solid, I'm just going to try to make sure that it's tested in CI before merging

**Comment by inge4pres** [comment]
Hi :) no worries at all! Thank you for getting in touch

**Comment by notzippy** [comment]
Any updates on this ? I have an application in need of this as well :-D

**Comment by nelsam** [comment]
I'm working on it now; just need to figure out how to get travis to spin up a crate DB.

File: dialect_crate.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+	return "DELETE FROM"
+}
+
+// Returns "?"
+func (d CrateDialect) BindVar(i int) string {
+	return "?"
+}
+
+func (d CrateDialect) QuoteField(f string) string {
+	return "'" + f + "'"
```
Comment (nelsam) [inline dialect_crate.go]: This is incorrect.  Crate uses double-quotes for fields and whatnot.  It should  be ``return `"` + f + `"` ``.

**Comment by nelsam** [comment]
I added crate to the tests and it has been failing in a few ways.  The first was the quoted field problem, but even after solving that, [it had failures](https://travis-ci.org/go-gorp/gorp/jobs/290936839).

I've pushed a commit to your fork that adds tests; can you work on getting the tests passing?

**Comment by inge4pres** [comment]
Hi @nelsam as this PR is 7 months old, I will reset my dev environment.
I tihnk I should also pull latest changes from master to consitently release the feature to add support for Crate.

From `travis.yml` I can see Crate is installed using the deprecated PPA installation: this forces use of 2.0.x, while a more recent version is available through adding repositories to have 2.1.x and higher, see [install-cratedb-from-ppa-legacy](https://crate.io/docs/crate/guide/en/latest/getting_started/install/local/linux.html#install-cratedb-from-ppa-legacy) and [install-from-deb-repo](https://crate.io/docs/crate/guide/en/latest/getting_started/install/local/linux.html#install-from-deb-repo).

**Comment by nelsam** [comment]
@inge4pres yeah, feel free to update the  `.travis.yml` in whatever way is best.  I have no experience with crate, I just pulled the  PPA from the install script (https://install.crate.io).

There are currently no conflicts with master, but feel free to merge or rebase on top of master first and do what you need to.

**Comment by inge4pres** [comment]
I went as far as making pass the majority of tests, but at a certain point I'm facing that CrateDB has no INDEX statement available.

**Comment by nelsam** [comment]
It's fine to update the tests to skip testing index against create. We've
had to do similar things with one of the mysql drivers.

See [here](https://github.com/go-gorp/gorp/blob/master/gorp_test.go#L738)
for an example.

**Comment by aman00323** [comment]
Any update on this ?

**Pull request closed without merging**